Sources/MapboxCoreNavigation/EHorizon/RoadObjectMatcher.swift

```diff
@@ -0,0 +1,113 @@
+                native.setListenerFor(self)
+            } else {
+                native.setListenerFor(nil)
+            }
+        }
+    }
+
+    /**
+     Matches given OpenLR object to the graph.
+
+     - parameter location: OpenLR location of the road object, encoded in a base64 string.
+     - parameter standard: Standard used to encode OpenLR location.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchOpenLR(location: String, standard: OpenLRStandard, identifier: RoadObjectIdentifier) {
+        native.matchOpenLR(forBase64Encoded: location, standard: standard.native, id: identifier)
+    }
+
+    /**
+     Matches given polyline to the graph.
+     Polyline should define a valid path on the graph,
+     i.e. it should be possible to drive this path according to traffic rules.
+
+     - parameter polyline: Polyline representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchPolyline(_ polyline: [CLLocation], identifier: RoadObjectIdentifier) {
+        native.matchPolyline(forPolyline: polyline, id: identifier)
+    }
+
+    /**
+     Matches a given polygon to the graph.
+     "Matching" here means we try to find all intersections of the polygon with the road graph
+     and track distances to those intersections as distance to the polygon.
+
+     - parameter polygon: Polygon representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchPolygon(_ polygon: [CLLocation], identifier: RoadObjectIdentifier) {
+        native.matchPolygon(forPolygon: polygon, id: identifier)
+    }
+
+    /**
+     Matches given gantry (i.e. polyline orthogonal to the road) to the graph.
+     "Matching" here means we try to find all intersections of the gantry with the road graph
+     and track distances to those intersections as distance to the gantry.
+
+     - parameter gantry: Gantry representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchGantry(_ gantry: [CLLocation], identifier: RoadObjectIdentifier) {
+        native.matchGantry(forGantry: gantry, id: identifier)
+    }
+
+    /**
+     Matches given point to road graph.
+
+     - parameter point: Point representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchPoint(_ point: CLLocationCoordinate2D, identifier: RoadObjectIdentifier) {
+        native.matchPoint(forPoint: point, id: identifier)
+    }
+
+    /**
+     Cancel road object matching.
+
+     - parameter identifier: Identifier for which matching should be canceled.
+     */
+    public func cancel(identifier: RoadObjectIdentifier) {
+        native.cancel(forId: identifier)
+    }
+
+    init(_ native: MapboxNavigationNative.RoadObjectMatcher) {
+        self.native = native
+    }
+
+    deinit {
+        native.setListenerFor(nil)
```

Comment:
Shouldn't NavNative handle this?

Reply:
Do you mean whether listeners are held weakly in NavNative? I'm not counting on that, but let's ask @SiarheiFedartsou